app/controllers/api/cloud_templates_controller.rb

```diff
@@ -8,6 +8,35 @@ def check_compliance_resource(type, id, _data = nil)
       end
     end
 
+    def import_resource(_type, _id, data = {})
+      params = %w(dst_provider_id src_provider_id src_image_id cos_provider_id cos_bucket_id)
+      raise BadRequestError, "Parameter 'data' has to contain non-empty values for the keys '#{params.join(", ")}', received: '#{data}'" if data.values_at(*params).any?(&:blank?)
+      raise BadRequestError, "Source and destination provider identifiers must differ"  if data['dst_provider_id'] == data['src_provider_id']
+
+      ems_dst = resource_search(data['dst_provider_id'], :providers, collection_class(:providers))
+      raise BadRequestError, "Specified destination provider cannot be found by this id: '#{data['dst_provider_id']}'" if ems_dst.nil?
+
+      ems_src = resource_search(data['src_provider_id'], :providers, collection_class(:providers))
+      raise BadRequestError, "Specified source provider cannot be found by this id: '#{data['src_provider_id']}'" if ems_src.nil?
+
+      src_image = resource_search(data['src_image_id'], :providers, collection_class(:templates))
+      raise BadRequestError, "Source image specified by the id '#{data['src_image_id']}' cannot be found in the source provider with id '#{ems_src.id}'" if src_image.nil? || src_image.ems_id != ems_src.id
+
+      cos = resource_search(data['cos_provider_id'], :providers, collection_class(:providers))
```

Comment:
Is the this different from the destination provider?  I thought the ibm cloud object storage manager was the target provider

Reply:
no, it is a transient storage, the final destination is the _ems_dst_.

Reply:
Hmm. But I see the problem now. If this API were to be generic it should probably not contain anything about the transient storage as it this step is specific to the PVS <----> PVC scenario. I guess I'll have to validate these parameters in PVS/PVC and not in the API controller.